### DISCUSSION TITLE
How to Metadata.Get<MyType>("property")

### DISCUSSION THREAD

**Discussion by JoshClose** [statiqdev/Discussions#49] [Q&A]
I have a file `_index.json` alongside my markdown file and I'm trying to get data out of it.

My json file looks like this.
```json
{
    "toc": [
        { "title": "the title", "link": "#the-link" }
    ]
}
```
My type looks like this.
```cs
public class TocItem
{
    public string Title { get; set; }
    public string Link { get; set; }
    public List<TocItem> Children { get; set; } = new List<TocItem>();
}
```

I tried different variations of getting values out.
```cs
var items = Metadata.Get<List<TocItem>>("toc");
var items = Metadata.Get<TocItem[]>("toc");
var items = Metadata.GetList<TocItem>("toc");
```

Is there a way to do this?

I'd be happy if I could get the raw JSON at "toc" and just use Newtonsoft to deserialize it.

**Comment by JoshClose** [comment]
I was able to do this.

```cs
var items = TocItem.Create(Metadata.GetList<IMetadataDictionary>("toc"));
```

```cs
public class TocItem
{
    public string Title { get; set; }

	public string Link { get; set; }

	public List<TocItem> Children { get; set; } = new List<TocItem>();

	public static TocItem Create(IMetadataDictionary data)
	{
		var tocItem = new TocItem()
		{
			Title = data.Get<string>("title"),
			Link = data.Get<string>("link"),
		};

		if (data.ContainsKey("children"))
		{
			tocItem.Children = Create(data.GetList<IMetadataDictionary>("children"));
		}

		return tocItem;
	}

	public static List<TocItem> Create(IEnumerable<IMetadataDictionary> objects) => objects?.Select(Create).ToList() ?? new List<TocItem>();
}
```

Not sure if this is the best way.

**Reply by daveaglick** [reply]
Manual binding like this is definitely how I would bind an object from front matter or a sidecar file right now. Unfortunately Statiq doesn't yet have a reflection-based binder.

Another option would be to escape the JSON string within the JSON so you're effectively setting a string in the JSON that just happens to itself contain escaped JSON, and then call `.GetString("toc")` (or `.Get<string>("toc")` - they're equivalent) and use a JSON deserializer.

This has come up before, but I don't seem to have an issue for it yet - created https://github.com/statiqdev/Statiq.Framework/issues/168
